texi2html/5.0/share/man/man1/texi2html.1
.TH TEXI2HTML 1 "30 June 2010"
.AT 3
.SH NAME
texi2html \- a Texinfo to HTML converter
.SH SYNOPSIS
.B texi2html [options] file
.SH DESCRIPTION
.I Texi2html
converts the given Texinfo file to a set of HTML files. It tries to handle
most of the Texinfo commands. It creates hypertext links for cross-references,
footnotes...
.PP
.I Texi2html
may furthermore use latex2html to generate HTML (code and/or images) for @math
and @iftex tags (see the 
.B \-\-l2h
option).
.PP
.I Texi2html
creates several files depending on the contents of the Texinfo file and on
the chosen options (see FILES).
.PP
The HTML files created by
.I texi2html
are in general closer to TeX than to Info. Using init files (see
the 
.B \-\-init\-file
option), other styles or output formats may be selected.
.SH OPTIONS
Options may be abbreviated to th shortests unique prefix. The
following is only an incomplete list of the available options: 
.TP 12
.B \-\-help [1|2]
Print usage instructions and list the most important, all relevant,
or all recognised (including obsolete) command-line options, and exit. 
.TP
.B \-\-split chapter|section|node
Split the output into several HTML files (one per main chapter/section or node).
.TP
.B \-\-ifinfo \-\-iftex \-\-ifxml \-\-ifplaintext \-\-ifhtml
Expand @ifinfo, @iftex and @tex, @ifxml and @xml, @ifplaintext, 
@ifhtml and @html regions, and don't expand @ifnot 
corresponding regions (default: ifhtml).
.TP
.B \-\-no\-ifinfo \-\-no\-iftex \-\-no\-ifxml \-\-no\-ifplaintext \-\-no\-html
Don't expand @ifinfo, @iftex, @ifxml, @ifplaintext regions and expand
@ifnot corresponding regions.
.TP
.B \-I \fIdir\fP
Append \fIdir\fP to the directory list searched for included files.
.TP
.B \-P \fIdir\fP
Prepend \fIdir\fP to the directory list searched for included files.
.TP
.B \-\-l2h
Use latex2html for content of @math and @tex tags. Content must be such that 
both, latex and tex, can translate it correctly. I.e., no tex-only constructs 
must be used.
.TP
.B \-\-css\-include \fIfile\fP
Use css file \fIfile\fP.
.TP
.B \-\-init\-file \fIfile\fP
Load init file \fIfile\fP (see also FILES).
.TP
.B \-\-document\-language \fIlang\fP
Use \fIlang\fP as document language.
.TP
.B \-\-macro\-expand \fIfile\fP
output macro expanded source in \fIfile\fP.
.TP
.B \-\-no\-menu
Don't show the Texinfo menus; by default they are shown.
.TP
.B \-\-no\-number
Don't number the sections.
.TP
.B \-\-use\-nodes
Use nodes for sectionning.
.TP
.B \-\-prefix
Use as prefix for output files, instead of basename.
.TP
.B \-\-node\-files
Produce one file per node for cross references.
.TP
.B \-\-l2h\-l2h \fIprog\fP  
Use \fIprog\fP as latex2html program (default \fIlatex2html\fP). 
Program must be executable. 
.TP
.B \-\-l2h\-init \fIfile\fP  
Use \fIfile\fP as latex2html init file. \fIfile\fP should be used to specify in which way latex2html handles mathematical expressions and formulas. 

.TP
.B \-\-l2h\-tmp \fIdir\fP
Use \fIdir\fP as tmp dir for latex2html. This directory (resp. its path) 
must not contain a dot  (i.e., a ".").
.TP 
.B \-\-l2h\-clean
remove all intermediate files generated by latex2html (prevents caching!)  
.TP 
.B \-\-l2h\-skip
Do not really run latex2html. Use (kept) files of previous runs, instead.
.TP
.B \-\-verbose
Give a verbose output. 
.TP
.B \-\-version
Print version and exit.
.PP
.SH FILES
Initialisation options are read first from /usr/local/Cellar/texi2html/5.0/share/texi2html/Config 
(the exact location being changeable with the
.RI --datadir=dir
option to the configure script), then from /usr/local/Cellar/texi2html/5.0/etc/texi2html/Config
(the exact location being changeable with the
.RI --sysconfdir=dir
option to the configure script), then from ./Config, from 
$HOME/.texi2html/Config, and from any command-line options including 
\-\-init\-file options; with later settings overriding earlier ones.
.P
By default
.I texi2html
my creates the following files (foo being the name of the Texinfo file):
.TP 16
.B foo_toc.html
The table of contents (if @contents is present).
.TP
.B foo.html
The document's contents or the Top element.
.TP
.B foo_fot.html
The footnotes (if any).
.TP
.B foo_abt.html
An help section.
.TP
.B foo_l2h_img*
Images generated by latex2html (if any).
.PP
When used with the
.B \-\-split
option, it creates several files (one per chapter or node), named
.B foo_n.html
(n being the index of the chapter or node), and the
.B foo.html
file contains the Top element.
.PP
Otherwise it creates only one file:
.B foo.html
.SH VERSION
This is \fItexi2html\fP version 5.0, 30 June 2010.
.SH AUTHOR
Texi2html was originally written by is Lionel Cons, CERN IT/DIS/OSE,
Lionel.Cons@cern.ch.  Over time, many other people around the net
contributed to this program.  It was maintained by Olaf Bachmann,
obachman@mathematik.uni-kl.de, and is now maintained by Derek Price.
.SH AVAILABILITY
The latest version of this program, as well as support in the form of mailing
lists with archives, bug tracking, documentation, and other tools can be found
at:
	 http://www.nongnu.org/texi2html/
.SH COPYRIGHT
Copyright \(co 1999-2005  Patrice Dumas <pertusus@free.fr>,
                        Derek Price <derek@ximbiot.com>,
                        Adrian Aichner <adrian@xemacs.org>,
                      & others.

There is NO warranty.  You may redistribute this software
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
.SH "SEE ALSO"
GNU Texinfo Documentation Format
.br
http://texinfo.org/
.br
World Wide Web (WWW),
HyperText Markup Language (HTML)
.br
http://www.w3.org
.br
LaTeX2HTML
.br
http://www-dsed.llnl.gov/files/programs/unix/latex2html
.SH BUGS
This man-page is often out-of-date. The info manual should be more up-to-date.
.ex


texi2html/5.0/.brew/texi2html.rb
class Texi2html < Formula
  desc "Convert TeXinfo files to HTML"
  homepage "http://www.nongnu.org/texi2html/"
  url "https://download.savannah.gnu.org/releases/texi2html/texi2html-5.0.tar.gz"
  sha256 "e60edd2a9b8399ca615c6e81e06fa61946ba2f2406c76cd63eb829c91d3a3d7d"

  keg_only :provided_pre_mountain_lion

  depends_on "gettext"

  def install
    system "./configure", "--disable-dependency-tracking", "--prefix=#{prefix}",
                          "--mandir=#{man}", "--infodir=#{info}"
    chmod 0755, "./install-sh"
    system "make", "install"
  end

  test do
    (testpath/"test.texinfo").write <<-EOS.undent
      @ifnottex
      @node Top
      @top Hello World!
      @end ifnottex
      @bye
    EOS
    system "#{bin}/texi2html", "test.texinfo"
    assert_match /Hello World!/, File.read("test.html")
  end
end
